pkg/controller/che/che_controller.go

```diff
@@ -448,7 +448,14 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 					return &reconcile.Result{Requeue: true, RequeueAfter: time.Second * 5}, err
 				}
 			}
-			if effectiveDeployment.Spec.Template.Spec.Containers[0].Image != registryImage {
+			actualMemRequest := effectiveDeployment.Spec.Template.Spec.Containers[0].Resources.Requests[corev1.ResourceMemory]
+			actualMemLimit := effectiveDeployment.Spec.Template.Spec.Containers[0].Resources.Limits[corev1.ResourceMemory]
+			limitStr := actualMemLimit.String()
+			requestStr := actualMemRequest.String()
+			if effectiveDeployment.Spec.Template.Spec.Containers[0].Image != registryImage ||
```

Comment:
why do we need these extra checks here?

Reply:
To identify situations when caracteristics of the current deployment are not the same as those required by the custom resource. In such a situation, we should redeploy => restart the POD.